add support for `sku_name` and `trace` for `azurerm_spring_cloud_service` (#7531)
terraform-providers/terraform-provider-azurerm: website/docs/r/spring_cloud_service.html.markdown

  location = "Southeast Asia"
}

resource "azurerm_application_insights" "example" {
  name                = "tf-test-appinsights"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  application_type    = "web"
}

resource "azurerm_spring_cloud_service" "example" {
  name                = "example-springcloud"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  sku_name            = "S0"

  config_server_git_setting {
    uri          = "https://github.com/Azure-Samples/piggymetrics"
    label        = "config"
    search_paths = ["dir1", "dir2"]
  }

  trace {

>     enabled                         = true

Reviewer: (as above) if we use the presence of the block to infer this we can make this:

```suggestion
```

>     app_insight_instrumentation_key = azurerm_application_insights.example.instrumentation_key

Reviewer: we can shorten this down to:

```suggestion
    instrumentation_key = azurerm_application_insights.example.instrumentation_key
```

* `strict_host_key_checking_enabled` - (Optional) Indicates whether the Config Server instance will fail to start if the host_key does not match.

---

The `trace` block supports the following:

* `enabled` - (Required) Indicates whether the Trace is enabled.

> * `app_insight_instrumentation_key` - (Optional) The instrumentationKey of Application Insights.

Reviewer: can we make this:

```suggestion
* `instrumentation_key` - (Optional) The Instrumentation Key used for Application Insights.
```
